src/multiset.rs

```diff
@@ -298,6 +327,193 @@ impl<T> Sub for HashMultiSet<T> where
     }
 }
 
+impl<T> SubAssign for HashMultiSet<T> where
+    T: Eq + Hash + Clone
+{
+    /// Remove the elements of one `HashMultiSet` from another
+    /// using `-`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use multiset::HashMultiSet;
+    /// use std::iter::FromIterator;
+    ///
+    /// let mut lhs: HashMultiSet<isize> = FromIterator::from_iter(vec![1,2,3]);
+    /// let rhs: HashMultiSet<isize> = FromIterator::from_iter(vec![1,1,4]);
+    /// lhs -= rhs;
+    /// assert_eq!(0, lhs.count_of(1));
+    /// assert_eq!(1, lhs.count_of(2));
+    /// assert_eq!(1, lhs.count_of(3));
+    /// assert_eq!(0, lhs.count_of(4));
+    /// ```
+    fn sub_assign(&mut self, rhs: HashMultiSet<T>) {
+        for val in rhs.distinct_elements() {
+            let count = rhs.count_of((*val).clone());
+            self.remove_times((*val).clone(), count);
+        }
+    }
+}
+
+
+impl<T> PartialEq for HashMultiSet<T> where
+    T: Eq + Hash + Clone
+{
+    /// The obvious definition of equality: exactly the
+    /// same number of the same elements.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use multiset::HashMultiSet;
+    /// use std::iter::FromIterator;
+    ///
+    /// let s1: HashMultiSet<isize> = FromIterator::from_iter(vec![1,2,3]);
+    /// let s2: HashMultiSet<isize> = FromIterator::from_iter(vec![1,1,4]);
+    /// assert!(s1 == s1);
+    /// assert!(s1 != s2);
+    /// ```
+    fn eq(&self, other: &HashMultiSet<T>) -> bool {
+        for val in self.distinct_elements() {
+            if self.count_of(val.clone()) != other.count_of(val.clone()) {
+                return false;
+            };
+        };
+        true
+    }
+}
+
+/// Test the degree to which the RHS is contained by the LHS.
+/// * A result of `Equal` means that the LHS exactly
+/// matches on all RHS elements.
+/// * A result of `Greater` means that the LHS contains
+/// all present RHS elements and overcontains at least one.
+/// * A result of `Less` means only that the containment
+/// does not hold.
+fn lcr<T: Eq + Hash + Clone>(lhs: &HashMultiSet<T>,
+                             rhs: &HashMultiSet<T>) -> Ordering
+{
+    let mut result = Ordering::Equal;
+    for val in rhs.distinct_elements() {
+        let nlhs = lhs.count_of(val.clone());
+        if nlhs == 0 {
+            return Ordering::Less;
+        };
+        let nrhs = rhs.count_of(val.clone());
+        if nrhs > nlhs {
+            return Ordering::Less;
+        } else if nrhs < nlhs {
+            result = Ordering::Greater;
+        };
+    };
+    result
+}
```

Comment:
What does `lcr` stand for? My gut reaction is this is such a specialized comparison operation that maybe it should be in its own crate or module. It seems a bit strange to impose a particular total ordering on multisets.